EZYVideoPlayer/Classes/EZYVideoPlayerModel.swift

```diff
@@ -0,0 +1,156 @@
+        let interval = CMTime(seconds: 0.5, preferredTimescale: CMTimeScale(NSEC_PER_SEC))
+        let mainQueue = DispatchQueue.main
+        // add the observer, using a closure as the callback function
+        timeObserver = player?.addPeriodicTimeObserver(forInterval: interval, queue: mainQueue, using: { [weak self] time in
+            guard let currentItem = self?.player?.currentItem else {return}
+            // call delegate method to pass the current time
+            self?.delegate?.playerDidChanged(position: Float(currentItem.currentTime().seconds))
+        })
+    }
+    
+    deinit {
+        player?.currentItem?.removeObserver(self, forKeyPath: PlayerObserverKey[.status])
+        player?.currentItem?.removeObserver(self, forKeyPath: PlayerObserverKey[.duration])
+        guard let t = timeObserver else {return}
+        player?.removeTimeObserver(t)
+        timeObserver = nil
+    }
+}
+
+//MARK: - Actions
+
+extension EZYVideoPlayerModel {
+    
+    func isPlaying() -> Bool {
+        shouldPlay = !shouldPlay
+        if shouldPlay {
+            player?.play()
+            delegate?.didChangedPlayer(status: .playing)
+        }else {
+            player?.pause()
+            delegate?.didChangedPlayer(status: .paused)
+        }
+        return shouldPlay
+    }
+    
+    func should(mute: Bool) {
+        player?.isMuted = mute
+    }
+    
+    func configureAs(menuItem: PlayerMenu) {
+        delegate?.didSelectMenu(item: menuItem)
+    }
+    
+    /// This function allows for seeking through a media player, such as a video or audio player. It takes in a value (withValue) as a parameter, representing the position in seconds that the player should seek to.
+    func seek(withValue: Float) {
+        // The player's seek method is called with a CMTime object created using the value and timescale of 1000.
+        player?.seek(to: CMTimeMake(value: Int64(withValue*1000), timescale: 1000))
+        // The playerDidChanged method of the delegate is called, passing in the position (withValue) as a parameter.
+        delegate?.playerDidChanged(position: withValue)
+    }
+    
+    /// This function allows for seeking forward in a media player. It calls the didChangedPlayer method on the delegate with the status "seekForward", then calls the seek function with a closure that calculates the new time to seek to.
+    func seekForward() {
+        // The didChangedPlayer method of the delegate is called with the status "seekForward".
+        delegate?.didChangedPlayer(status: .seekForward)
+        // The seek function is called with a closure that calculates the new time to seek to by adding 10 seconds
+        // to the current time. The closure ensures that the new time is not more than 10 seconds from the end of the duration
+        seek { duration, currentTime in
+            let newTime = currentTime + 10.0
+            return newTime < (duration - 10) ? newTime : duration
+        }
```

Comment:
Please add the value of `10` in the constant file, so that you don't have to put hardcode value in every time.